core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -229,4 +229,11 @@ private[spark] object UI {
     .stringConf
     .transform(_.toUpperCase(Locale.ROOT))
     .createWithDefault("LOCAL")
+
+  val UI_SQL_GROUP_SUB_EXECUTION_ENABLED = ConfigBuilder("spark.ui.sql.groupSubExecutionEnabled")
```

Comment:
Maybe, the following is better because the following introduce no new config namespace.
```scala
+ spark.ui.sql.groupSubExecutionEnabled
- spark.ui.groupSQLSubExecutionEnabled
```